docs/programming/os.md

```diff
@@ -0,0 +1,87 @@
+# Working with and without OS
+
+To squeeze as much performance as possible, ACE allows for disabling Amiga OS and re-enabling it when needed.
+Typically, 95% of stuff which game performs doesn't need operating system, so you're safe to have it disabled when your game runs.
+Things that require having OS enabled:
+
+- memory allocation and deallocation - most `ace/managers/memory.h` functions
+- logging to file - all `ace/managers/log.h` functions if log file is created and when in debug mode
+- general file access - all `ace/utils/file.h` and `ace/utils/dir.h` functions
+- system .library and .device usage
+- multithreading, communicating with other apps
+
+## Manually managing OS state
+
+**By default, ACE leaves OS in enabled state.**
+You should disable it using `systemUnuse()` to reach the maximum performance.
+
+The functions which require OS will try to re-enable and disable it on the fly.
+Although it's safe approach, it is not optimal, since each OS enabling/disabling takes lots of time (about 2-4 frames, or 40-80ms).
+Consider following code:
+
+```c
+systemUnuse(); // Assume that OS is disabled at this point
+
+logWrite("I'm initializing stuff\n");
+paletteLoad("palette.plt", s_pPalette, 32);
+s_pBm1 = bitmapCreate(320, 256, 5);
+s_pBm2 = fontCreate("test.fnt");
+logWrite("I'm done initializing\n");
+```
+
+This small code wastes about quarter to half a second to do very simple things, because each of this function re-enables OS at start and disables it at its end.
+This can be easily fixed by adding `systemUse()` and `systemUnuse()` calls:
+
+```c
+systemUse(); // Re-enable OS once
+logWrite("I'm initializing stuff\n");       // Any of these functions
+paletteLoad("palette.plt", s_pPalette, 32); // Won't re-enable/disable OS
+s_pBm1 = bitmapCreate(320, 256, 5);         // They have systemUse()/systemUnuse calls
+s_pBm2 = fontCreate("test.fnt");            // But they won't do anything since OS
+logWrite("I'm done initializing\n");        // Is already enabled
+systemUnuse(); // Disable OS once when we're done with it
+```
+
+## Best practices
+
+Try to have OS disabled at the end of gamestate creation, and to re-enable it at the beginning of gamestate destruction.
+This way, the game loop performs with maximum performance, whereas you have OS at your disposal in init/deinit phases.
```

Comment:
```suggestion
This way, the game loop performs with maximum performance, whereas you have OS at your disposal in create/destroy phases.
```
It's better to keep existings terms instead of providing new ones in docs.